### PR TITLE
Add range operator in Remote Inbox Notification

### LINKED ISSUE #44787: Remote Spec Rules: Add `range` operator
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

**Issue Description:**

Add a new Remote Spec operator ([README](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Admin/RemoteInboxNotifications/README.md)) for rules such that a rule becomes truthy if a derived value falls within a range of integers. Below would be a usage example.

```
"is_visible": {
  "type": "option",
  "value": [ a, b ],
  "default": 0,
  "operation": "range",
  "option_name": "woocommerce_remote_variant_assignment",
}
```
Be sure to guard against errors or fatals when dealing with type errors or undefined values.

### PR DESCRIPTION
**Pull request by moon0326**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #44787 

This PR adds `range` operation in Remote Inbox Notification to support testing a number's range. This PR also updates Remote Spec Validation schema files to support `range`

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Run unit tests and make sure they pass. `./vendor/bin/phpunit --testsuite shard2`
2. Install and activate WooCommerce Beta Tester
3. Go to `Tools -> WCA Test Helper -> Remote Spec Rule Validator`
4. Get the value of `woocommerce_remote_variant_assignment` option. 
5. Paste the following rule and adjust `value` for testing.

```
[
    {
        "type": "option",
        "option_name": "woocommerce_remote_variant_assignment",
        "value": [0, 10],
        "operation": "range"
    }
]
```

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [x] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

Support range operator in Remote Inbox Notification 

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/src/Admin/RemoteInboxNotifications/ComparisonOperation.php [hunk 1]
```diff
@@ -64,6 +64,8 @@ public static function compare( $left_operand, $right_operand, $operation ) {
 					return strpos( $left_operand, $right_operand ) === false;
 				}
 				break;
+			case 'range':
+				return $left_operand >= $right_operand[0] && $left_operand <= $right_operand[1];
```
Comment (chihsuan) [inline plugins/woocommerce/src/Admin/RemoteInboxNotifications/ComparisonOperation.php:71]: Should we check if `$right_operand` is an array and has at least 2 elements like what we have in other cases?

**Review by chihsuan — APPROVED** [review]
Tested well. 👍  Just left a minor comment about the code.

Pre-approving!

File: plugins/woocommerce/src/Admin/RemoteInboxNotifications/README.md [hunk 2]
```diff
@@ -183,6 +188,22 @@ onboarding profile:
 }
 ```
 
+#### range
+
+`range` operator can check if a number falls within a certain range
```
Comment (psealock) [inline plugins/woocommerce/src/Admin/RemoteInboxNotifications/README.md]: This would be a good place to note its `inclusive`

**Review by psealock — APPROVED** [review]
Looks great! Just one minor thing

File: plugins/woocommerce/src/Admin/RemoteInboxNotifications/ComparisonOperation.php (hunk 1 above)
Comment (moon0326) [reply plugins/woocommerce/src/Admin/RemoteInboxNotifications/ComparisonOperation.php:71]: @chihsuan Good suggestion! I'll make the change.

File: plugins/woocommerce/src/Admin/RemoteInboxNotifications/README.md (hunk 2 above)
Comment (moon0326) [reply plugins/woocommerce/src/Admin/RemoteInboxNotifications/README.md]: @psealock Good suggestion! I'll make the change.

File: plugins/woocommerce/src/Admin/RemoteInboxNotifications/ComparisonOperation.php (hunk 1 above)
Comment (moon0326) [reply plugins/woocommerce/src/Admin/RemoteInboxNotifications/ComparisonOperation.php:71]: Updated in https://github.com/woocommerce/woocommerce/pull/45201/commits/d48f340724e180725548a5dfcb0d80a835de99a2

File: plugins/woocommerce/src/Admin/RemoteInboxNotifications/README.md (hunk 2 above)
Comment (moon0326) [reply plugins/woocommerce/src/Admin/RemoteInboxNotifications/README.md]: Updated in https://github.com/woocommerce/woocommerce/pull/45201/commits/f9b3cf2ce9dcf15a80aa136a92f8f77601fa4c77

**Pull request merged**